mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -3256,6 +3325,114 @@ impl MmCoin for TendermintCoin {
             .await
     }
 
+    /// Overrides the default `pre_check_for_order_creation` implementation with
+    /// additional IBC-related logic on top of the default behavior.
+    #[cfg(feature = "ibc-routing-for-swaps")]
+    async fn pre_check_for_order_creation(
+        &self,
+        ctx: &MmArc,
+        rel_coin: &crate::MmCoinEnum,
+    ) -> MmResult<(), crate::OrderCreationPreCheckError> {
+        use crate::{lp_coinfind, MmCoinEnum, OrderCreationPreCheckError};
+
+        /// Looks for a Tendermint platform coin by the given ticker.
+        ///
+        /// Returns `Ok(Some(...))` if the coin exists and is a Tendermint platform coin,
+        /// `Ok(None)` if it's not active, or an error if somethings goes wrong or the ticker
+        /// isn't belongs to a Tendermint platform coin.
+        async fn find_tendermint_platform_coin(
+            ctx: &MmArc,
+            ticker: &str,
+        ) -> Result<Option<TendermintCoin>, MmError<OrderCreationPreCheckError>> {
+            match lp_coinfind(ctx, ticker).await {
+                Ok(Some(MmCoinEnum::Tendermint(coin))) => Ok(Some(coin)),
+                Ok(Some(other)) => MmError::err(OrderCreationPreCheckError::InternalError {
+                    reason: format!(
+                        "Expected a Tendermint coin for '{}', but found '{}'.",
+                        ticker,
+                        other.ticker()
+                    ),
+                }),
+                Ok(None) => Ok(None),
+                Err(reason) => MmError::err(OrderCreationPreCheckError::PreCheckFailed { reason }),
+            }
+        }
+
+        /// Picks an HTLC coin (IRIS or NUCLEUS) based on which IBC channel is configured
+        /// and is healthy.
+        async fn get_htlc_coin(
+            coin: &TendermintCoin,
+            ctx: &MmArc,
+        ) -> Result<Option<TendermintCoin>, MmError<OrderCreationPreCheckError>> {
+            const IRIS_PREFIX: &str = "iaa";
+            const IRIS_TICKER: &str = "IRIS";
+
+            const NUCLEUS_PREFIX: &str = "nuc";
+            const NUCLEUS_TICKER: &str = "NUCLEUS";
+
+            if coin.get_healthy_ibc_channel_for_address(IRIS_PREFIX).await.is_ok() {
+                return find_tendermint_platform_coin(ctx, IRIS_TICKER).await;
+            }
+
+            if coin.get_healthy_ibc_channel_for_address(NUCLEUS_PREFIX).await.is_ok() {
+                return find_tendermint_platform_coin(ctx, NUCLEUS_TICKER).await;
+            }
```

Comment:
hmmm, since this is checked here but we might get a race condition if status changes before usage..

Reply:
`get_healthy_ibc_channel_for_address` is a RO operation. We send a request to the remote node and read the response. There is no way to fall into race condition.